### ISSUE TITLE
Add support for AWS ECR in gitops v2 templates

### ISSUE THREAD

**Issue by NissesSenap** [XenitAB/azure-devops-templates#75]
We are working on supporting AWS in XKF and as a part of that we should add support for AWS in gitopsv2 templates.

**Comment by NissesSenap** [comment]
I'm trying to just have one main.yaml file that supports two push solutions, one for AWS and one for Azure.
I'm unable to solve this using if statments, for some reason the if don't become true.

Here is a example of the implementation that I have done:

```yaml
parameters:
  - name: poolVmImage
    type: string
    default: "ubuntu-16.04"
  - name: poolName
    type: string
    default: ""
  - name: environments
    type: object
  - name: azureSubscriptionTemplate
    type: string
  - name: acrNameTemplate
    type: string
    default: ""
  - name: imagePathPrefix
    type: string
    default: ""
  - name: gitopsPromotionTag
    type: string
    default: "v0.0.6"
  - name: awsRegion
    type: string
  - name: group
    type: string
    default: ""

stages:
  - stage: push_azure
    jobs:
      - ${{ each env in parameters.environments }}:
        - ${{ if eq(parameters.awsRegion, '') }}:
          - template: push-azure.yaml
            parameters:
              poolVmImage: ${{ parameters.poolVmImage }}
              poolName: ${{ parameters.poolName }}
              environment: ${{ env.name }}
              azureSubscription: ${{ format(parameters.azureSubscriptionTemplate, env.name) }}
              acrName: ${{ format(parameters.acrNameTemplate, env.name) }}
              imagePathPrefix: ${{ parameters.imagePathPrefix }}
      - job: Workaround1
        steps:
        - script: echo ignore, this is only to trick azure devops

  - stage: push_aws
    jobs:
      - ${{ each env in parameters.environments }}:
        - ${{ if eq(parameters.awsRegion, '') }}:
          - template: push-aws.yaml
            parameters:
              poolVmImage: ${{ parameters.poolVmImage }}
              poolName: ${{ parameters.poolName }}
              environment: ${{ env.name }}
              serviceConnection: ${{ format(parameters.azureSubscriptionTemplate, env.name) }}
              awsRegion: ${{ parameters.awsRegion }}
      - job: Workaround2
        steps:
        - script: echo ignore, this is only to trick azure devops

  - stage: new
    dependsOn: [push_aws, push_azure]
    jobs:
      - template: new.yaml
        parameters:
          poolVmImage: ${{ parameters.poolVmImage }}
          poolName: ${{ parameters.poolName }}
          gitopsPromotionTag: ${{ parameters.gitopsPromotionTag }}
          group: ${{ parameters.group }}
```

To move forward I will create a second new folder containing the AWS config.
Pros:
* Less logic needed

Cons:
* More yaml to manage

Note the `job: WorkaroundX` this is created else we get
```txt
Stage push_azure must contain at least one job with no dependencies
```
If anyone have suggestions on how to make this work it would be great.

**Comment by NissesSenap** [comment]
The reason why I tried to solve this in the main file was due to that even if I used a condition to define that task the Azure task shouldn't be run it still validated the serviceConnection string and in my case I was using a AWS service connection and then the Azure task verification failed.

I might have been to smart for my own good and tried to use one variable.
If i instead use two variabels, one for AWS service connection and one for Azure and just leave it empty the Azure one might not do a verification.

And will give this a quick test tomorrow. If that is the case I will probably go back to using just one folder compared to what I have currently done in the PR.
